src/git_provider/mod.rs

```diff
@@ -47,11 +48,27 @@ pub trait GitProvider {
     fn set_token(&self, path: &PathBuf) -> Result<String>;
     fn get_token(&self) -> Result<String>;
 
+    fn set_title(&self, branch: &String) -> Result<String> {
```

Comment:
For me, this looks more light a getter, than a setter.

```suggestion
    fn get_title(&self, branch: &String) -> Result<String> {
```

Reply:
I think neither name is that fitting, as a setter shouldn't return a value and a getter should just retrieve it, not set it. But I guess you could say get_title is shorthand for get_title_from_user

Reply:
Then you can name it ask_pr_title. But the default should be the commit title.